starter-code/scripts/views/articleView.js

```diff
@@ -38,6 +41,10 @@
   };
 
   // COMMENT: What does this method do?  What is it's execution path?
+  //We think there is a type here, and we want "on"change instead of "one"change, but...
+  //This function is called by articleController, which is called as the callback on any page load besides the about and admin pages.
+  //This function is activated when the user changes the value of either filter.
+  //This funciton gets the value of the filter choice and plugs it into the url.
```

Comment:
"one" actually is the correct method, it's not a typo. The difference is that the event handler fires once, then the event listener is removed. This is necessary because the function that adds the event listener (handleFilters) is getting called repeatedly, every time the articles page is re-rendered. With 'on', it would be stacking multiple event listeners onto the same select tags, and the event listener would fire multiple times as it's selected. Try changing the 'one' to 'on', running the code and using the filters for a minute. You'll see the site get really slow and making extraneous calls to the database.